ReportUtilityBills/tests/utility_bills_test.rb

```diff
@@ -363,66 +412,1081 @@ def test_monthly_prorate
+    @hpxml.pv_systems.each { |pv_system| pv_system.max_power_output = 1000.0 / @hpxml.pv_systems.size }
+    @hpxml.header.utility_bill_scenarios.each do |utility_bill_scenario|
+      fuels = _load_timeseries('../tests/PV_1kW.csv', utility_bill_scenario)
+      utility_rates, utility_bills = @measure.setup_utility_outputs()
+      _bill_calcs(fuels, utility_rates, utility_bills, @hpxml.header, @hpxml.pv_systems, utility_bill_scenario)
+      assert(File.exist?(@bills_csv))
+      actual_bills = _get_actual_bills(@bills_csv)
+      @expected_bills['Test: Electricity: Fixed ($)'] = 108
+      @expected_bills['Test: Electricity: Marginal ($)'] = 9892 # FIXME
+      @expected_bills['Test: Electricity: PV Credit ($)'] = -112
+      expected_bills = _get_expected_bills(@expected_bills)
+      _check_bills(expected_bills, actual_bills)
+    end
+  end
+
+  def test_detailed_calculations_sample_real_time_pricing_pv_10kW_net_metering_user_specified_excess_rate_min_annual_charge
+    @hpxml.header.utility_bill_scenarios[-1].elec_tariff_filepath = '../../ReportUtilityBills/tests/Sample Real-Time Pricing Rate Min Annual Charge.json'
+    @hpxml.pv_systems.each { |pv_system| pv_system.max_power_output = 10000.0 / @hpxml.pv_systems.size }
+    @hpxml.header.utility_bill_scenarios.each do |utility_bill_scenario|
+      fuels = _load_timeseries('../tests/PV_10kW.csv', utility_bill_scenario)
+      utility_rates, utility_bills = @measure.setup_utility_outputs()
+      _bill_calcs(fuels, utility_rates, utility_bills, @hpxml.header, @hpxml.pv_systems, utility_bill_scenario)
+      assert(File.exist?(@bills_csv))
+      actual_bills = _get_actual_bills(@bills_csv)
+      @expected_bills['Test: Electricity: Fixed ($)'] = 108
+      @expected_bills['Test: Electricity: Marginal ($)'] = 9892 # FIXME
+      @expected_bills['Test: Electricity: PV Credit ($)'] = -393
+      expected_bills = _get_expected_bills(@expected_bills)
+      _check_bills(expected_bills, actual_bills)
+    end
+  end
+
+  def test_downloaded_utility_rates
+    # skip # this test takes a long time and shouldn't be part of regular ci testing
+
+    require 'zlib'
+    require 'tempfile'
+
+    tar_extract = Gem::Package::TarReader.new(Zlib::GzipReader.open(File.join(File.dirname(__FILE__), '../resources/data/detailed_rates/rates.tar.gz')))
+    tar_extract.rewind # The extract has to be rewinded after every iteration
+    tar_extract.each do |entry|
+      next unless entry.file?
+
+      puts entry.full_name
+
+      tmpdir = Dir.tmpdir
+      tmpfile = Tempfile.new(['rate', '.json'], tmpdir)
+      tmp_path = tmpfile.path.to_s
+
+      File.open(tmp_path, 'wb') do |f|
+        f.print entry.read
+
+        @hpxml.header.utility_bill_scenarios.each do |utility_bill_scenario|
+          utility_bill_scenario.elec_tariff_filepath = tmp_path
+          fuels = _load_timeseries('../tests/PV_None.csv', utility_bill_scenario)
```

Comment:
Why can't you do this once up front?

Reply:
I am. I put a "skip" in there to prevent it from running on CI.

Reply:
Oh, sorry. I re-read. You mean literally the `_load_timeseries` line?

Reply:
Yes. Should only need to call it a single time in the method, no?